framework/src/utils/MooseMeshXYCuttingUtils.C

```diff
@@ -710,15 +721,17 @@ quadToTriOnLine(ReplicatedMesh & mesh,
       for (unsigned int i = 0; i < 4; i++)
       {
         const Point v_point = (*elem_it)->point(i);
-        node_side_rec.push_back(lineSideDeterminator(v_point(0),
-                                                     v_point(1),
-                                                     cut_line_params[0],
-                                                     cut_line_params[1],
-                                                     cut_line_params[2],
-                                                     true));
+        if (!pointOnLine(
+                v_point(0), v_point(1), cut_line_params[0], cut_line_params[1], cut_line_params[2]))
+          node_side_rec.push_back(lineSideDeterminator(v_point(0),
+                                                       v_point(1),
+                                                       cut_line_params[0],
+                                                       cut_line_params[1],
+                                                       cut_line_params[2],
+                                                       true));
       }
-      if (std::accumulate(node_side_rec.begin(), node_side_rec.end(), 0) != 4 &&
-          std::accumulate(node_side_rec.begin(), node_side_rec.end(), 0) > 0)
+      const auto num_nodes = std::accumulate(node_side_rec.begin(), node_side_rec.end(), 0);
```

Comment:
add a comment please

this is counting the booleans in the node_side_rec, which does not include nodes that were exactly on the line (excluded from the decision)